packages/tree-widget/src/components/TreeWidgetComponent.tsx

```diff
@@ -0,0 +1,210 @@
+  const spatialTreeProps = props.config?.spatialTree;
+  const storyTreeProps = props.config?.storyTree;
+  const realityDataProps = props.config?.realityData;
+
+  const modelsTreeComponent = React.useCallback(
+    () => (
+      <ModelsTreeComponent
+        iModel={imodel}
+        activeView={activeView}
+        enablePreloading={enablePreloading}
+        {...modelsTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, modelsTreeProps]
+  );
+
+  const categoriesTreeComponent = React.useCallback(
+    () => (
+      <CategoriesTreeComponent
+        iModel={imodel}
+        activeView={activeView}
+        enablePreloading={enablePreloading}
+        {...categoriesTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, categoriesTreeProps]
+  );
+
+  const spatialContainmentComponent = React.useCallback(
+    () => (
+      <SpatialContainmentTree
+        iModel={imodel}
+        enablePreloading={enablePreloading}
+        {...spatialTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, spatialTreeProps]
+  );
+
+  const storiesTreeComponent = React.useCallback(
+    () => (
+      <StoriesTree
+        iModel={imodel}
+        enablePreloading={enablePreloading}
+        {...storyTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, storyTreeProps]
+  );
+
+  const realityDataComponent = React.useCallback(
+    () => (
+      <RealityDataWidget
+        iModel={imodel}
+        enablePreloading={enablePreloading}
+        accessToken={props.accessToken}
+        {...realityDataProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, realityDataProps]
+  );
+
+  const [renderedElement, setRenderedElement] = useState<JSX.Element>(
+    modelsTreeComponent
+  );
+  const themedSelectOptions = [
+    {
+      label: UiFramework.translate("visibilityWidget.modeltree"),
+      value: "model-tree",
+    },
+
+    {
+      label: UiFramework.translate("visibilityWidget.categories"),
+      value: "categories-tree",
+    },
+
+    {
+      label: UiFramework.translate("visibilityWidget.containment"),
+      value: "spatial-containment-tree",
+    },
+
+    {
+      label: TreeWidget.translate("stories"),
+      value: "stories-tree",
+    },
+
+    {
+      label: TreeWidget.translate("realityData"),
+      value: "reality-data-tree",
+    },
+  ];
+
+  return (
+    <div className="tree-widget-ui-test-app-visibility-widget">
+      <div className="components-selectable-content">
+        <div className="components-selectable-content-header">
+          <ThemedSelect
+            isSearchable={false}
+            options={themedSelectOptions}
+            defaultValue={themedSelectOptions[0]}
+            onChange={(newValue: any, actionMeta: any) => {
+              console.log("value:", newValue, actionMeta);
+              if (actionMeta?.action === "select-option" && newValue) {
+                switch ((newValue as OptionType).value) {
```

Comment:
Console.log could be removed...?

@zjzoltek, you can commit this as-is, but we want to replace ThemedSelect with _SelectableContent_ from ui-components.  SelectableContent uses the HTML select, and after speaking with Grigas, he agrees we should swap that with ThemedSelect.  Let's add this to the backlog, and if you don't feel comfortable making a change in ui-components, I can add it to my backlog.

Reply:
Fixed spacing and added default case.

@JoeZMan I think I'll go ahead and commit as-is and later on down the road come back and replace the `ThemedSelect` with `SelectableContent`. Main reason is because I don't want to hold this PR any longer (I still have to move reality data from ui-snippets to viewer-components-react before I can get this through). I'm comfortable with making the change in ui-components. Not sure if you've created a backlog item yet, but if you have go ahead and assign it to me.